airflow/providers_manager.py

```diff
@@ -278,6 +309,51 @@ def _get_attr(obj: Any, attr_name: str):
             return None
         return getattr(obj, attr_name)
 
+    def _add_taskflow_decorator(
+        self, decorator_name, decorator_class_name: str, provider_package: str
+    ) -> None:
+        if provider_package.startswith("apache-airflow"):
+            provider_path = provider_package[len("apache-") :].replace("-", ".")
+            if not decorator_class_name.startswith(provider_path):
+                log.warning(
+                    "Sanity check failed when importing '%s' from '%s' package. It should start with '%s'",
+                    decorator_class_name,
+                    provider_package,
+                    provider_path,
+                )
+                return
+        if decorator_name in self._taskflow_decorator_dict:
+            log.warning(
+                "The hook_class '%s' has been already registered.",
+                decorator_class_name,
+            )
+            return
+        try:
+            module, class_name = decorator_class_name.rsplit('.', maxsplit=1)
+            decorator_class = getattr(importlib.import_module(module), class_name)
+            self._taskflow_decorator_dict[decorator_name] = decorator_class
```

Comment:
This causes _all_ decorators to be imported up front, which was something I wanted us to avoid -- i.e. I'd like `airflow.providers.docker.decorators.docker.docker_decorator` to only be imported on the first access (per process) of `@task.docker`.

Reply:
As discussed before - I will make another change/refactor to the "providers_manager" to accomplish that in "generic" way

Reply:
@ashb so is this something that should be addressed in this PR or another? As it stands there won't be many (if any) decorators to import, so maybe we can defer?

Reply:
@potiuk did you ever make a refactor for this? If not I'd like to figure out a way we can defer this issue so we can get this merged.

Reply:
@ashb I think we can kick the can on this. For now there will be all of zero decorators that will be loaded in this case. I think we can reasonably assume that we're not going to see a flood of custom decorators in the near future. This can be a 2.2.1 feature etc.